### PR TITLE
[WPB-700] Pass along server errors between Galley and Stern

### PR DESCRIPTION
**Pull request by elland**
https://wearezeta.atlassian.net/browse/WPB-700

## Checklist

 - [ ] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [ ] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: services/brig/src/Brig/API/Internal.hs [hunk 1]
```diff
@@ -72,7 +72,7 @@ import Data.Qualified
 import Data.Set qualified as Set
 import Data.Time.Clock.System
 import Imports hiding (head)
-import Network.Wai.Routing hiding (toList)
+import Network.Wai.Routing hiding (toList, trace)
```
Comment (fisx) [inline services/brig/src/Brig/API/Internal.hs:75]:
```suggestion
import Network.Wai.Routing hiding (toList)
```

File: tools/stern/src/Stern/Intra.hs [hunk 2]
```diff
@@ -454,18 +455,18 @@ getTeamBillingInfo :: TeamId -> Handler (Maybe TeamBillingInfo)
...
         ( method GET
             . Bilge.paths ["i", "team", toByteString' tid, "billing"]
         )
-  case Bilge.statusCode r of
-    200 -> Just <$> parseResponse (mkError status502 "bad-upstream") r
+  case Bilge.statusCode resp of
+    200 -> Just <$> parseResponse (mkError status502 "bad-upstream") resp
     404 -> pure Nothing
-    _ -> throwE (mkError status502 "bad-upstream" "bad response")
+    _ -> throwE (mkError status502 "bad-upstream" (errorMessage resp))
```
Comment (fisx) [inline tools/stern/src/Stern/Intra.hs:469]: i thought the idea was to re-throw the upstream error message?  isn't that more helpful than wrapping that in a 5xx?

**Review by fisx — COMMENTED** [review]
looks reasonable, but of course i have comments.  :)  also: are there more end-points in stern that need this treatment?

File: tools/stern/src/Stern/Intra.hs (hunk 2 above)
Comment (elland) [reply tools/stern/src/Stern/Intra.hs:469]: I thought we just didn't want to lose the meta information from the error response, but still re-throw upstream issues as 502s.

**Comment by elland** [comment]
> looks reasonable, but of course i have comments. :) also: are there more end-points in stern that need this treatment?

Didn't find any others :)

**Review by fisx — APPROVED** [review]

**Pull request merged**